### DISCUSSION TITLE
src/App/Http/Livewire directory does not exist

### DISCUSSION THREAD

**Discussion by ju5t** [livewire/livewire#2369] [1. Help] [answered]
I'm not opening this as a bug because it's related to our setup.

We're using domains. Our app lives in `src/App`, everything else in `src/Domain`. This works on a local environment where we use Valet. When we deploy this app to a server with Forge it errors out in some cases with the error:

`The "src/App/Http/Livewire" directory does not exist`

This happens when:

- We include a Livewire component in a blade view: `<livewire:component />`
- An XHR call runs and triggers a re-render.

It does not happen when:

- We load a component through a route: `Route::get('login', Login::class)->name('auth.login');`, where `Login::class` is a Livewire component. When we interact with the component, the error does show.

The error originates from: 

https://github.com/livewire/livewire/blob/956d1b5032ae3570122a106091b9757bedd6e78e/src/LivewireComponentsFinder.php#L63-L77

This eventually calls this function:

https://github.com/symfony/finder/blob/5.x/Finder.php#L578-L596

Which eventually triggers the exception.

The folder does exit. What is going wrong here?

**Comment by ju5t** [comment, marked as answer]
I found the discrepancy. When I removed `bootstrap/cache/livewire-components.php` from my local environment I'm getting the same error as on production.

**Reply by ju5t** [reply]
And the solution is `php artisan livewire:discover`. Added this to our Forge deployment and everything's good to go now!

**Reply by bilogic** [reply]
@ju5t i get the same error when running `php artisan livewire:discover`. I have my livewire `app.blade.php` is in a package as `package-app.blade.php`, any idea how to handle it? Thanks!

**Reply by adamhaley** [reply]
> 

I was having the same problem. I just manually created the directory and ran php artisan livewire:discover again

**Reply by joshhanley** [reply]
This should be fixed in the next release, so the directory won't need to exist. See PR #4923
